### DISCUSSION TITLE
Extend curl_easy_getinfo() to get the mechanism used to authenticate with server or proxy

### DISCUSSION THREAD

**Discussion by genotrance** [curl/curl#12668] [Ideas]
When `CURLOPT_PROXYAUTH` / `CURLOPT_HTTPAUTH` is set to `ANY` or similar, there's no way to know which mechanism libcurl used to authenticate. Knowing this would help reduce the round trip on subsequent connections.

`curl_easy_getinfo()` includes `CURLINFO_HTTPAUTH_AVAIL` and `CURLINFO_PROXYAUTH_AVAIL` which gives you all the available options returned by the server but we don't know which one was used once auth is successful.

**Comment by genotrance** [comment]
Bumping this thread for visibility.

**Comment by bagder** [comment]
A quite reasonable idea I think. I'll give it a go.

**Reply by bagder** [reply]
See #15450 for work in progress
